config/initializers/foreman.rb

```diff
@@ -42,8 +42,16 @@
 Foreman::Plugin.medium_providers_registry.register MediumProviders::Default
 
 Rails.application.config.to_prepare do
-  # Force reload settings after all plugins have loaded and on code reload
-  Foreman.settings.load if (Setting.table_exists? rescue(false))
+  if (Setting.table_exists? rescue(false))
+    # Force reload settings after all plugins have loaded and on code reload
+    Dir[
+      File.join(Rails.root, "app/models/setting.rb"),
+      File.join(Rails.root, "app/models/setting/*.rb"),
```

Comment:
Not sure if it works (perhaps it needs `.to_s`) but isn't this cleaner, relying on the fact `Rails.root` is a [pathname](https://ruby-doc.org/stdlib-2.6.3/libdoc/pathname/rdoc/Pathname.html):
```suggestion
      Rails.root.join('app', 'models', 'setting.rb'),
      Rails.root.join('app', 'models', 'setting', '*.rb'),
```

I also see we have the same implementation around line 20. Is this a static method we should implement on `Settings`?

Reply:
I agree with slight improvement, if we are ok to change both, but no static method for this uglieness please.
We want to get rid of it by finishing the Setting dsl and migrating from STI to this DSL